core/controllers/admin.py

```diff
@@ -288,7 +288,12 @@ def post(self):
                 result = {
                     'version': version
                 }
-            elif action == 'update_feature_flag_rules':
+            # The handler schema defines the possible values of 'action'.
```

Comment:
I think unit tests should cover all code paths that might arise, but the `else` condition here is guaranteed to never occur because of our schema validators (which we have/will have tests for). Therefore, I don't see the point in writing a test for the `else` condition.

I can see how you might be concerned that if we make a mistake in the code, the `else` condition might actually happen. In that case, I think the solution is to throw an exception in the `else` case rather than right a test for it. Does that make sense?

Reply:
Hmm, I think an exception might be better solution, also I think it might be possible to use this approach https://hakibenita.com/python-mypy-exhaustive-checking to catch these in the MyPy.

Also, the problem that I see is that it is quite hard to understand why we do this and where we use what. I think there should be some common documented approach for handling the if, elif, …, elif, else branching when backend testing.